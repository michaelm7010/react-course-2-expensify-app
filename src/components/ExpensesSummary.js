import React from 'react';
import numeral from 'numeral';
import { connect } from 'react-redux';
import selectExpenses from '../selectors/expenses';
import selectExpensesTotal from '../selectors/expenses-total';

const formatExpenses = (expensesTotal) => 
    numeral(expensesTotal / 100).format('$0,0.00');

const expenseWord = (expenseCount) => expenseCount === 1 ? 'expense' : 'expenses';
 
export const ExpensesSummary = ({ expenseCount, expensesTotal }) => (
    <div>
        {
            expenseCount === 0 ? (
                <h1>No expenses meeting filter criteria</h1>
            ) : (
                <h1>Viewing {expenseCount} {expenseWord(expenseCount)} totaling {formatExpenses(expensesTotal)}</h1>
            )                
        }
    </div>
);

const mapStateToProps = (state) => {
    const visibleExpenses = selectExpenses(state.expenses, state.filters);
    return {
        expenseCount:  visibleExpenses.length,
        expensesTotal:  selectExpensesTotal(visibleExpenses)
    };
};

export default connect(mapStateToProps)(ExpensesSummary);
