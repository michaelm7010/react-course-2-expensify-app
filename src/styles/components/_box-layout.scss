.box-layout {
    align-items:  center;
    background:  url('/images/bg.jpg');
    // always cover the entire window
    background-size:  cover;
    display: flex;
    // set height, width to 100 percent of viewport height, width
    height:  100vh;
    justify-content:  center;
    width:  100vw;
}

.box-layout__box {
    background:  fade-out(white, .15);
    border-radius:  3px;
    // large padding on top and bottom, medium patting on left and right
    padding:  $l-size $m-size;  
    text-align:  center;
    width:  25rem;
}

.box-layout__title {
    // create margin for bottom
    margin:  0 0 $m-size 0;
    // ensure space between lines
    line-height:  1;
}