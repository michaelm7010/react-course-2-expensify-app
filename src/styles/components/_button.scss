.button {
    background:  $blue;
    border:  none;
    color:  white;
    display:  inline-block;
    font-size:  $font-size-large;
    font-weight:  300;  
    // button always one-line
    line-height:  1;
    padding:  $s-size;
    text-decoration:  none;
}

.button--link {
    background:  none;
}

// set grey background for secondary buttons
.button--secondary {
    background:  #888888;
}