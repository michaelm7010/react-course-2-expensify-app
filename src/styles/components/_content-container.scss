// this style ensures that content on pages is centered on the web page
.content-container {
    // auto even distributes space on left and right
    margin:  0 auto;
    max-width:  80rem;
    // medium padding on left, right side
    padding:  0 $m_size;
}