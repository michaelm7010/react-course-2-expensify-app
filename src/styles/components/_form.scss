.form {
    display:  flex;
    flex-direction:  column;
    // target changes on all descendants 
    >  * {
        margin-bottom:  $m-size;
    }
}

.form__error {
    // set margin to all sides except bottom
    margin:  0 0 $l_size 0;
    font-style:  italic;
}