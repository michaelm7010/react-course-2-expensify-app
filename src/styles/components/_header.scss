.header {
    background:  $dark-blue;
}

.header__content {
    // center items vertically
    align-items:  center;
    display:  flex;
    // distribute leftover space between title and button
    justify-content:  space-between;
    // pad small space on top, botton, zero on sides
    padding:  $s-size 0;
}

.header__title {
    color:  white;
    // remove underscore from link
    text-decoration:  none;
    // specify margin for all h1 components in this class
    h1 {
        margin:  0;
    }
}