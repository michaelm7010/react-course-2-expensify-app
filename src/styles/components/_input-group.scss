.input-group {
    display:  flex;
    flex-direction:  column;
    margin-bottom:  $s_size;

    // these are styles to be used for larger screens (overrides smaller screens values above)
    @media (min-width:  $desktop-breakpoint) {
        flex-direction:  row;
        margin-bottom:  $l-size;
    }
}

.input-group__item {
    margin-bottom:  $s-size;
    // these are styles to be used for larger screens (overrides smaller screens values above)
    @media (min-width:  $desktop-breakpoint) {
        // margin on right
        margin:  0 $s-size 0 0
    }
}