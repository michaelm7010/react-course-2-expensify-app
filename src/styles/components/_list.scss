.list-header {
    background:  $off-white;
    border:  1px solid darken($off-white, 7%);
    color:  $grey;
    display:  flex;
    // distribute leftover space between items
    justify-content:  space-between;
    padding:  $s-size $m-size;
}

.list-body {
    margin-bottom:  $m-size;
    @media (mind-width:  $desktop-breakpoint) {
        margin-bottom:  $l-size;
    }

}

.list-item {
    border:  1px solid darken($off-white, 7%);
    // remove border on top because the one above it already has a border there
    border-top:  none;
    color:  $dark-grey;
    display:  flex;
    // on mobile display, we want things going in columns
    flex-direction:  column;
    padding:  $s-size;
    text-decoration:  none;
    // transition defines three things:  property we wish to transition, duration of transition, function we wish to use
    transition:  background .3s ease;
    &:hover {
        background:  $off-white;
    }  

    // override properties for desktop
    @media (min-width:  $desktop-breakpoint) {
        // center items vertically
        align-items:  center;
        flex-direction:  row;
        justify-content:  space-between;
        padding:  $m-size;
    }
}

.list-item--message {
    align-items:  center;
    color:  $grey;
    justify-content:  center;
    padding:  $m-size;
    &:hover {
        background:  none;
    }
}

.list-item__title {
    margin:  0;
    // in the case of a long word without spaces, this wraps the lines
    word-break:  break-all;
}

.list-item__subtitle {
    color:  $grey;
    font-size:  $font-size-small;
}

.list-item__data {
    // for mobile, we don't want much margin on top or bottom of amount
    margin:  $s-size 0 0 0;
    // override properties for desktop
    @media (min-width:  $desktop-breakpoint) {
        margin:  0;
        padding-left:  $s-size;
    }
}