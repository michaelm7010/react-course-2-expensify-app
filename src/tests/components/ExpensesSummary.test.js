import React from 'react';
import { shallow } from 'enzyme';
import { ExpensesSummary } from '../../components/ExpensesSummary';
import expenses from '../fixtures/expenses';

test('should render expense summary with no expenses', () => {
    const wrapper = shallow(<ExpensesSummary expenseCount={0} expensesTotal={0}/>);
    expect(wrapper).toMatchSnapshot();
});
test('should render expense summary with 1 expense', () => {
    const wrapper = shallow(<ExpensesSummary expenseCount={1} expensesTotal={234}/>);
    expect(wrapper).toMatchSnapshot();
});

test('should render expense summary with multiple expenses', () => {
    const wrapper = shallow(<ExpensesSummary expenseCount={12} expensesTotal={1234}/>);
    expect(wrapper).toMatchSnapshot();
});