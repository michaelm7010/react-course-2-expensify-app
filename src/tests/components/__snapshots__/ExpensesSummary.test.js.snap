// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render expense summary with 1 expense 1`] = `
<div>
  <h1>
    Viewing 
    1
     
    expense
     totaling 
    $2.34
  </h1>
</div>
`;

exports[`should render expense summary with multiple expenses 1`] = `
<div>
  <h1>
    Viewing 
    12
     
    expenses
     totaling 
    $12.34
  </h1>
</div>
`;

exports[`should render expense summary with no expenses 1`] = `
<div>
  <h1>
    No expenses meeting filter criteria
  </h1>
</div>
`;
